### ISSUE TITLE
Colouring particles?

### ISSUE THREAD

**Issue by AhmedSalih3d** [tkeskita/BVtkNodes#9]
Hello!

Now that time selector is working (thanks btw) I am trying to a render of my simulation inside of Blender. For me, my simulation data is particles (which using this package is imported as mesh vertices), which looks like;

![image](https://user-images.githubusercontent.com/36305327/75536287-ba9ebe00-5a15-11ea-8002-71e7b8354319.png)

My problem is now that, I want to animate these particles by using spheres etc. I am able to do this by the following approach;

1. Select my object
2. Add particle filter
3. Use hair
4. Insert number of hairs to be exactly equal to number of vertices
5. Under "Source" emit from "Verts"
6. Under "Source" disable random order
7. Under "Render" select object and choose any mesh (for example a Sphere)

Then I can get something like this;

![image](https://user-images.githubusercontent.com/36305327/75536769-1ec18200-5a16-11ea-9bb6-847d112a6acf.png)

Where the "orange spheres" are a moving piston, and the "black vertices" is the fluid. My problem now arises;

I can do the exact same thing for the Fluid particles, but I have no clue of how to color the spheres based on their data. What I have been trying is;

![image](https://user-images.githubusercontent.com/36305327/75536923-59c3b580-5a16-11ea-8ad7-79b1d8ef3eb6.png)

But this would only be able to color the "black vertices" and I am not sure if it possible to color / scale up vertices in Blender.

So basically my question is, is there a hack / workflow to parse this velocity data to the particle info in the Shading settings?

What I seek is something like this;

![image](https://user-images.githubusercontent.com/36305327/75537047-97c0d980-5a16-11ea-9797-c2cf44f543c0.png)

Where the fluid particles are coloured by velocity. This was done in Paraview. 

Kind regards

**Comment by tkeskita** [comment]
Hi, currently color info goes via surfaces from VTK to Blender. So currently you must create geometry and color it in VTK. I put [example here](https://blenderartists.org/t/bvtknodes-gallery/1161079/22). If you have many particles it will be slow, so I'm gonna think and test if it is possible to pass points and vertex colors from VTK to Blender. That way you could utilize particle instancing in Blender, which is far more efficient for even medium size visualizations. I'll reply here if I can make it happen (might take a while).

**Comment by AhmedSalih3d** [comment]
Thanks!

I have tried to implement your example, but unfortunately I cannot make the "Custom Code" options appear? Am I doing something wrong?

![image](https://user-images.githubusercontent.com/36305327/75605231-492c4180-5ae1-11ea-8d19-62c5ea1e8737.png)

But it works great with scalar values, like idp (id number of each particle) and pressure! Idp shown down below. As you said unfortunately this is too slow for animations as of now, but it is nice to atleast be able to do it.

For anyone having trouble to render, just give it a few tries, it might crash the first time. Here is the picture I rendered, very excited about this possibility. Hope you are able to get the more efficient approach to work.

![TestRender](https://user-images.githubusercontent.com/36305327/75605400-dd4ad880-5ae2-11ea-9719-9d0d964acb5d.png)


Kind regards

**Comment by tkeskita** [comment]
Hi, see [here](https://blenderartists.org/t/bvtknodes-gallery/1161079/19). The buttons for editing and saving custom code are in the Properties tab. You need to use Blender Text Editor. I'm going to write documentation, meanwhile that blenderartists thread is worth reading through.

**Comment by AhmedSalih3d** [comment]
Thanks for answering, I forgot too.

Will close it now
